src/ucp/core/ucp_worker.c

```diff
@@ -249,7 +249,13 @@ static void ucp_ep_err_pending_purge(uct_pending_req_t *self, void *arg)
     ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
     ucs_status_t status = (uintptr_t)arg;
 
-    ucp_request_complete_send(req, status);
+    if (req->send.uct_comp.func) {
+        if (--req->send.uct_comp.count == 0) {
```

Comment:
single fragment zero-copy does not increment counter on posting, the same like last fragment on multi zero-copy. I don't think that this non-consistency of UCP should be fixed in UCT. UCT can't know that 2 completions requested per 1 real operation. Otherwise, we need to have possibility to extract requests from the pending queue. This is not trivial for any transport.
In additional, we have stub ep which is implemented in UCP.